### PR TITLE
JAMES-3088 Favor composition for UsersRepository

### PR DESCRIPTION
**Pull request by chibenwa**
Follow up of https://github.com/linagora/james-project/pull/3174

We may consider similar changes for RRT, and domain lists as well.

### CODE REVIEW

**Comment by chibenwa** [comment]
> It looks like you still use inheritance

Yes I use it to cheat Spring.

Otherwize I can 100% adopt composition like I did for Cassandra

**Comment by Arsnael** [comment]
```
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR] Errors: 
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   AuthenticatedCassandraJamesServerTest$AuthenticationTest.connectIMAPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   AuthenticatedCassandraJamesServerTest$AuthenticationTest.connectLMTPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   AuthenticatedCassandraJamesServerTest$AuthenticationTest.connectOnSecondaryIMAPServerIMAPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   AuthenticatedCassandraJamesServerTest$AuthenticationTest.connectPOP3ServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   AuthenticatedCassandraJamesServerTest$AuthenticationTest.connectSMTPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   AuthenticatedCassandraJamesServerTest$AuthenticationTest.hostnameShouldBeRetrievedWhenRestarting » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   AuthenticatedCassandraJamesServerTest$AuthenticationTest.hostnameShouldBeUsedAsDefaultDomain » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraJamesServerTest.connectIMAPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraJamesServerTest.connectLMTPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraJamesServerTest.connectOnSecondaryIMAPServerIMAPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraJamesServerTest.connectPOP3ServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraJamesServerTest.connectSMTPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraJamesServerTest.copyBatchSizeShouldEqualsConfigurationValue » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraJamesServerTest.hostnameShouldBeRetrievedWhenRestarting » Creation Un...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraJamesServerTest.hostnameShouldBeUsedAsDefaultDomain » Creation Unable...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraJamesServerTest.moveBatchSizeShouldEqualsConfigurationValue » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraJmapJamesServerTest.connectJMAPServerShouldRespondBadRequest » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraMailRepositoryIntegrationTest.deniedSenderMailShouldBeStoredInCassandraMailRepositoryWhenConfigured » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraNodeConfTest$NormalBehaviour.serverShouldStartServiceWhenNodeIsReachable:79 » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraNodeConfTest$OneFailedNode.serverShouldStartServiceWhenNodeIsReachable:100 » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraNodeConfTest$UseMappedPort.configShouldWorkWithNonDefaultPort:118 » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraWithTikaTest.connectIMAPServerShouldSendShabangOnConnect » Creation U...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraWithTikaTest.connectLMTPServerShouldSendShabangOnConnect » Creation U...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraWithTikaTest.connectOnSecondaryIMAPServerIMAPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraWithTikaTest.connectPOP3ServerShouldSendShabangOnConnect » Creation U...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraWithTikaTest.connectSMTPServerShouldSendShabangOnConnect » Creation U...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraWithTikaTest.hostnameShouldBeRetrievedWhenRestarting » Creation Unabl...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraWithTikaTest.hostnameShouldBeUsedAsDefaultDomain » Creation Unable to...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   DefaultCassandraJamesServerTest.cassandraJamesServerShouldStartWithNoConfigurationFile » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   ESReporterTest.timeMetricsShouldBeReportedWhenImapCommandsReceived » Creation ...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   ESReporterTest.timeMetricsShouldBeReportedWhenJmapRequestsReceived » Creation ...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   JamesServerWithRetryConnectionTest.serverShouldRetryToConnectToCassandraWhenStartService:89->assertThatServerStartCorrectly:113 » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   JamesServerWithRetryConnectionTest.serverShouldRetryToConnectToElasticSearchWhenStartService:98->assertThatServerStartCorrectly:113 » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   JamesServerWithRetryConnectionTest.serverShouldStartAtDefault:80->assertThatServerStartCorrectly:113 » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.connectIMAPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.connectJMAPServerShouldRespondBadRequest » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.connectLMTPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.connectOnSecondaryIMAPServerIMAPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.connectPOP3ServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.connectSMTPServerShouldSendShabangOnConnect » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.hostnameShouldBeRetrievedWhenRestarting » Creation ...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.hostnameShouldBeUsedAsDefaultDomain » Creation Unab...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.mailsContentWithUnicodeCharactersShouldBeKeptUnChanged » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.mailsShouldBeWellReceived » Creation Unable to crea...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   WithCassandraBlobStoreTest.oneHundredMailsShouldBeWellReceived » Creation Unab...
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraSchemaVersionStartUpCheckTest.serverShouldStartSuccessfullyWhenBetweenMinAndMaxVersion:104 » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraSchemaVersionStartUpCheckTest.serverShouldStartSuccessfullyWhenMaxVersion:94 » Creation
[1bc47b597be3b6de25e712da4aa94ef05b6e51f5] [ERROR]   CassandraSchemaVersionStartUpCheckTest.serverShouldStartSuccessfullyWhenMinVersion:114 » Creation
```
Something wrong with Cassandra guice bindings?

File: server/data/data-jpa/src/main/java/org/apache/james/user/jpa/JPAUsersDAO.java [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+     *         or if the password is incorrect
+     * 
+     * @since James 1.2.2
+     */
+    @Override
+    public boolean test(Username name, String password) throws UsersRepositoryException {
+        final User user = getUserByName(name);
+        final boolean result;
+        result = user != null && user.verifyPassword(password);
+        return result;
```
Comment (blackheaven) [inline server/data/data-jpa/src/main/java/org/apache/james/user/jpa/JPAUsersDAO.java]: `getUserByName` should return an `Optional` an rewrite this function

**Comment by chibenwa** [comment]
> RRTs and many other abstract-based classes hierarchy 

One step at a time?

I 100% agree

**Comment by chibenwa** [comment]
Force pushed to drop no review commit (and solve conflicts)

**Comment by chibenwa** [comment]
> RRTs and many other abstract-based classes hierarchy

(@blackheaven looks like what inter sprints are made for )

**Comment by Arsnael** [comment]
```
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR] Errors: 
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.addUserShouldAddAUserWhenEmptyRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.addUserShouldAddAUserWhenNotEmptyRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.addUserShouldDisableCaseVariationWhenOriginalValueLowerCased » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.addUserShouldDisableCaseVariation » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.addUserShouldThrowWhenSameUsernameWithDifferentCase » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.addUserShouldThrowWhenUserAlreadyPresentInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.containsShouldBeCaseInsentiveWhenOriginalValueLowerCased » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.containsShouldBeCaseInsentive » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.containsShouldPreserveCaseVariation » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.countUsersShouldReturnNumberOfUsersWhenNotEmptyRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.getUserByNameShouldBeCaseInsentive » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.getUserByNameShouldReturnAUserWhenContainedInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.getUserByNameShouldReturnLowerCaseAddedUser » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.getUserByNameShouldReturnUserWhenDifferentCase » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.listShouldReturnExactlyUsersInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.listShouldReturnLowerCaseUser » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.removeUserShouldBeCaseInsentiveOnCaseVariationUser » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.removeUserShouldBeCaseInsentive » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.removeUserShouldRemoveAUserWhenPresentInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.testShouldBeCaseInsentiveOnCaseVariationUser » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.testShouldBeCaseInsentive » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.testShouldReturnFalseWhenAUserHasAnIncorrectCasePassword » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.testShouldReturnFalseWhenAUserHasAnIncorrectPassword » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.testShouldReturnFalseWhenAUserIsNotInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.testShouldReturnFalseWhenAUserIsRemovedFromRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.testShouldReturnTrueWhenAUserHasACorrectPassword » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.testShouldReturnTrueWhenAUserHasAnIncorrectCaseName » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.updateUserShouldAllowToAuthenticateWithNewPassword » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.updateUserShouldNotAllowToAuthenticateWithOldPassword » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenDisableVirtualHosting.updateUserShouldThrowWhenAUserIsNoMoreInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.addUserShouldAddAUserWhenEmptyRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.addUserShouldAddAUserWhenNotEmptyRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.addUserShouldDisableCaseVariationWhenOriginalValueLowerCased » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.addUserShouldDisableCaseVariation » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.addUserShouldThrowWhenSameUsernameWithDifferentCase » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.addUserShouldThrowWhenUserAlreadyPresentInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.containsShouldBeCaseInsentiveWhenOriginalValueLowerCased » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.containsShouldBeCaseInsentive » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.containsShouldPreserveCaseVariation » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.countUsersShouldReturnNumberOfUsersWhenNotEmptyRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.getUserByNameShouldBeCaseInsentive » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.getUserByNameShouldReturnAUserWhenContainedInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.getUserByNameShouldReturnLowerCaseAddedUser » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.getUserByNameShouldReturnUserWhenDifferentCase » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.listShouldReturnExactlyUsersInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.listShouldReturnLowerCaseUser » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.removeUserShouldBeCaseInsentiveOnCaseVariationUser » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.removeUserShouldBeCaseInsentive » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.removeUserShouldRemoveAUserWhenPresentInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.testShouldBeCaseInsentiveOnCaseVariationUser » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.testShouldBeCaseInsentive » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.testShouldReturnFalseWhenAUserHasAnIncorrectCasePassword » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.testShouldReturnFalseWhenAUserHasAnIncorrectPassword » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.testShouldReturnFalseWhenAUserIsNotInRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.testShouldReturnFalseWhenAUserIsRemovedFromRepository » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.testShouldReturnTrueWhenAUserHasACorrectPasswordAndOtherCaseInDomain » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.testShouldReturnTrueWhenAUserHasACorrectPassword » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.testShouldReturnTrueWhenAUserHasAnIncorrectCaseName » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.updateUserShouldAllowToAuthenticateWithNewPassword » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.updateUserShouldNotAllowToAuthenticateWithOldPassword » AlreadyExistInUsersRepository
[69f071507aac5ca26e357adb41dbc4f5b126c28d] [ERROR]   CassandraUsersRepositoryTest$WhenEnableVirtualHosting.updateUserShouldThrowWhenAUserIsNoMoreInRepository » AlreadyExistInUsersRepository
```

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**